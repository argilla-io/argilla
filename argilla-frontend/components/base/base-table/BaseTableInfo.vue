<!--
  - coding=utf-8
  - Copyright 2021-present, the Recognai S.L. team.
  -
  - Licensed under the Apache License, Version 2.0 (the "License");
  - you may not use this file except in compliance with the License.
  - You may obtain a copy of the License at
  -
  -     http://www.apache.org/licenses/LICENSE-2.0
  -
  - Unless required by applicable law or agreed to in writing, software
  - distributed under the License is distributed on an "AS IS" BASIS,
  - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  - See the License for the specific language governing permissions and
  - limitations under the License.
  -->
<template>
  <transition appear name="fade">
    <div class="table-info">
      <div class="table-info__header">
        <slot name="columns">
          <div class="table-info__item">
            <div
              v-for="(column, key) in columns"
              :key="key"
              :class="[`table-info__item__col`, column.class]"
            >
              <lazy-table-filtrable-column
                :column="column"
                :data="data"
                :filters="filters"
                v-if="column.filtrable"
                @applyFilters="onApplyFilters"
              />
              <button
                v-else-if="column.sortable"
                :data-title="column.tooltip"
                :class="[sortOrder, { active: sortedBy === column.field }]"
                @click="sort(column)"
              >
                <svgicon width="18" height="18" color="#4D4D4D" name="sort" />
                <span>{{ column.name }}</span>
              </button>
              <button v-else :data-title="column.tooltip">
                <span>{{ column.name }}</span>
              </button>
            </div>
          </div>
        </slot>
      </div>
      <results-empty v-if="tableIsEmpty" :title="emptySearchInfo.title" />
      <template v-else>
        <div class="table-info__body" ref="table">
          <ul>
            <li v-for="item in filteredResults" :key="item.id" :id="item.id">
              <nuxt-link :to="rowLink(item)" class="table-info__item">
                <span
                  v-for="(column, idx) in columns"
                  :key="idx"
                  :class="[`table-info__item__col`, column.class]"
                >
                  <span :class="column.class">
                    <span v-if="column.actions">
                      <div class="table-info__actions">
                        <p
                          class="table-info__main"
                          v-if="column.type === 'main'"
                        >
                          {{ itemValue(item, column) }}
                        </p>
                        <span v-else>{{ itemValue(item, column) }}</span>
                        <div class="table-info__actions__buttons">
                          <base-action-tooltip
                            v-for="(action, idx) in column.actions"
                            :key="idx"
                            :tooltip="action.tooltip"
                          >
                            <base-button
                              :title="action.title"
                              class="table-info__actions__button button-icon"
                              @click.prevent="
                                onActionClicked(action.name, item)
                              "
                            >
                              <svgicon
                                v-if="action.icon !== undefined"
                                :name="action.icon"
                                width="16"
                                height="16"
                              />
                            </base-button>
                          </base-action-tooltip>
                        </div>
                      </div>
                    </span>
                    <base-date
                      v-else-if="column.type === 'date'"
                      format="date-relative-now"
                      :date="itemValue(item, column)"
                    />
                    <nuxt-link v-else-if="column.link" :to="column.link(item)">
                      {{ itemValue(item, column) }}
                    </nuxt-link>
                    <span v-else>{{ itemValue(item, column) }}</span>
                    <span v-if="column.component">
                      <component
                        v-if="hydrate[item.id]"
                        :is="column.component.name"
                        v-bind="{ ...column.component.props(item) }"
                      />
                    </span>
                  </span>
                </span>
              </nuxt-link>
            </li>
          </ul>
        </div>
      </template>
    </div>
  </transition>
</template>

<script>
import "assets/icons/trash-empty";
import "assets/icons/refresh";
import "assets/icons/copy";
import "assets/icons/settings";
import "assets/icons/link";
import "assets/icons/sort";

export default {
  props: {
    data: {
      type: Array,
      default: () => [],
    },
    columns: {
      type: Array,
      required: true,
    },
    sortedOrder: {
      type: String,
      default: "desc",
    },
    searchOn: {
      type: String,
      required: true,
    },
    sortedByField: {
      type: String,
      default: undefined,
    },
    emptySearchInfo: {
      title: undefined,
    },
    querySearch: {
      type: String,
      default: undefined,
    },
    activeFilters: {
      type: Array,
      default: () => {
        return [];
      },
    },
    rowLink: {
      type: Function,
      default: () => {
        return () => {};
      },
    },
  },
  data() {
    return {
      sortOrder: this.sortedOrder,
      sortedBy: this.sortedByField,
      filters: {},
      hydrate: {},
    };
  },
  watch: {
    filteredResults() {
      this.changeVisibility();
    },
  },
  computed: {
    tableIsEmpty() {
      return this.filteredResults && this.filteredResults.length === 0;
    },
    filterActions() {
      return this.actions;
    },
    filteredResults() {
      const matchSearch = (item) => {
        if (this.querySearch === undefined) {
          return true;
        }
        const querySearch = this.querySearch.toLowerCase();

        return item[this.searchOn]
          .toString()
          .toLowerCase()
          .includes(querySearch);
      };
      const matchFilters = (item) => {
        if (this.filters) {
          return Object.keys(this.filters).every((key) => {
            return this.filters[key].includes(item[key]);
          });
        }

        return true;
      };
      const itemComparator = (a, b) => {
        const modifier = this.sortedOrder === "desc" ? -1 : 1;

        if (a[this.sortedBy] < b[this.sortedBy]) return -1 * modifier;
        if (a[this.sortedBy] > b[this.sortedBy]) return 1 * modifier;

        return 0;
      };

      return this.data
        .filter(matchSearch)
        .filter(matchFilters)
        .sort(itemComparator);
    },
  },
  beforeMount() {
    this.sortedBy = this.sortedByField;

    return this.activeFilters
      .filter((filter) => filter.values.length)
      ?.forEach(({ column, values }) => {
        this.$set(this.filters, column, values);
      });
  },
  methods: {
    itemValue(item, column) {
      if (column.subfield) {
        return item[column.field][column.subfield];
      }
      return item[column.field];
    },
    onActionClicked(action, item) {
      this.$emit("onActionClicked", action, item);
    },
    sort(column) {
      this.$emit("sort-column", column.field, this.sortOrder);
      this.sortedBy = column.field;
      if (column.field === this.sortedBy) {
        this.sortOrder = this.sortOrder === "asc" ? "desc" : "asc";
      }
    },
    onApplyFilters(column, selectedOptions) {
      if (selectedOptions.length) {
        this.$set(this.filters, column.field, selectedOptions);
      } else {
        this.$delete(this.filters, column.field);
      }
      this.$emit("filter-applied", {
        column: column.field,
        values: selectedOptions,
      });
    },
    changeVisibility() {
      const handleIntersection = (entries) => {
        for (const entry of entries) {
          if (entry.isIntersecting) {
            this.$set(this.hydrate, entry.target.id, true);
          }
        }
      };

      const observer = new IntersectionObserver(handleIntersection);

      this.data.forEach((item) => {
        const element = document.getElementById(item.id);
        if (!element) return;

        observer.observe(element);
      });
    },
  },
  mounted() {
    this.changeVisibility();
  },
};
</script>

<style lang="scss" scoped>
$headerColor: palette(grey, 700);
$borderColor: palette(grey, 600);

.table-info {
  $this: &;
  display: flex;
  flex-direction: column;
  padding: 0;
  min-height: 0;
  list-style: none;
  color: $black-54;
  ul {
    list-style: none;
    padding: 0;
    margin: 0;
  }
  &__header {
    border: 1px solid $borderColor;
    border-bottom: 0;
    border-radius: var(--m, 10px) var(--m, 10px) 0px 0px;
    background: $headerColor;
    min-height: 49px;
    position: relative;
    margin-top: $base-space * 2;
    .table-info:has(.empty) & {
      border-bottom: 1px solid $borderColor;
    }
    &__checkbox {
      margin: 0 !important;
    }
    &__button {
      position: absolute !important;
      top: 2em;
      right: 2em;
      margin-bottom: 0 !important;
    }
    #{$this}__item {
      border: unset;
      min-height: 50px;
      background: none;
      padding-top: 0.2em;
      padding-bottom: 0.2em;
      display: flex;
      align-items: center;
    }
    button:not(.button) {
      cursor: pointer;
      border: 0;
      outline: none;
      background: transparent;
      padding-left: 0;
      padding-right: 0;
      color: $black-87;
      @include font-size(14px);
      text-align: left;
      display: flex;
      align-items: center;
      @include media("<=desktop") {
        display: block;
        .svg-icon {
          display: block;
        }
      }
      span {
        white-space: nowrap;
      }
    }
  }
  &__body {
    overflow: auto;
    padding-bottom: 0.5em;
    min-height: 250px;
    @extend %hide-scrollbar;
    #{$this}__item {
      &:hover,
      &:focus {
        background: #fcfcfc;
      }
    }
  }
  &__item {
    position: relative;
    display: flex;
    align-items: center;
    background: palette(white);
    list-style: none;
    padding: $base-space * 2 $base-space * 2;
    width: 100%;
    min-height: $base-space * 10;
    text-decoration: none;
    outline: none;
    border: 1px solid $borderColor;
    border-top: 0;

    &__col {
      text-align: left;
      margin-right: 1em;
      flex: 1 1 0px;
      &:nth-last-of-type(-n + 1) {
        max-width: 120px;
      }
      &:first-child {
        width: auto;
        min-width: auto;
        flex-grow: 2.5;
      }
      &.progress {
        min-width: 160px;
      }
    }
    .svg-icon {
      margin-right: $base-space;
      fill: $black-37;
      &:hover {
        fill: $black-54;
      }
    }
  }
  .empty {
    margin-top: 5em;
    height: auto;
    min-height: 50vh;
  }
  &__main {
    margin: 0;
    color: $black-87;
    .table-info__item:hover & {
      color: $primary-color;
    }
  }
  &__actions {
    display: flex;
    justify-content: flex-start;
    flex-direction: row;
    align-items: center;
    gap: $base-space;

    &__buttons {
      display: flex;
      flex-shrink: 0;

      &__button {
        position: relative;
        display: inline-block;
        .svg-icon {
          margin: 0;
        }
        & + #{$this} {
          margin-left: auto;
        }
      }
    }
  }
  &__title {
    display: block;
    hyphens: auto;
    word-break: break-word;
    .button-icon {
      padding: $base-space;
      display: inline-block;
      .svg-icon {
        margin: 0;
      }
    }
    a {
      text-decoration: none;
      &:hover {
        color: palette(black);
      }
    }
  }
  .text {
    color: $black-54;
    p {
      display: inline-block;
      background: palette(grey, 800);
      padding: 0.5em;
      border-radius: 10px;
      margin-right: 0.5em;
      margin-top: 0;
      hyphens: auto;
      word-break: break-word;
      &:last-child {
        margin-bottom: 0;
      }
    }
  }
}
button[data-title] {
  position: relative;
  @extend %has-tooltip--top;
  @extend %tooltip-large-text;
}
</style>
