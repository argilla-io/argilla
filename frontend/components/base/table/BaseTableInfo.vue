<!--
  - coding=utf-8
  - Copyright 2021-present, the Recognai S.L. team.
  -
  - Licensed under the Apache License, Version 2.0 (the "License");
  - you may not use this file except in compliance with the License.
  - You may obtain a copy of the License at
  -
  -     http://www.apache.org/licenses/LICENSE-2.0
  -
  - Unless required by applicable law or agreed to in writing, software
  - distributed under the License is distributed on an "AS IS" BASIS,
  - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  - See the License for the specific language governing permissions and
  - limitations under the License.
  -->

<template>
  <transition appear name="fade">
    <div class="table-info">
      <div class="table-info__header">
        <slot name="columns">
          <div class="table-info__item">
            <div
              v-for="(column, key) in columns"
              :key="key"
              class="table-info__item__col"
            >
              <lazy-table-filtrable-column
                :column="column"
                :data="data"
                :filters="filters"
                v-if="column.filtrable"
                @applyFilters="onApplyFilters"
              />
              <button
                v-else-if="column.sortable"
                :data-title="column.tooltip"
                :class="[sortOrder, { active: sortedBy === column.field }]"
                @click="sort(column)"
              >
                <svgicon width="18" height="18" color="#4D4D4D" name="sort" />
                <span>{{ column.name }}</span>
              </button>
              <button v-else :data-title="column.tooltip">
                <span>{{ column.name }}</span>
              </button>
            </div>
          </div>
        </slot>
      </div>
      <results-empty v-if="tableIsEmpty" :title="emptySearchInfo.title" />
      <template v-else>
        <div class="table-info__body">
          <ul>
            <li v-for="item in filteredResults" :key="String(item.id)">
              <div class="table-info__item">
                <base-checkbox
                  v-if="globalActions"
                  v-model="item.selectedRecord"
                  class="list__item__checkbox"
                  :value="item.name"
                  @change="onCheckboxChanged($event, item.id)"
                />
                <span
                  v-for="(column, idx) in columns"
                  :key="idx"
                  class="table-info__item__col"
                >
                  <span :class="column.class">
                    <a
                      v-if="column.type === 'action'"
                      href="#"
                      @click.prevent="onActionClicked(item.kind, item)"
                      >{{ itemValue(item, column) }}
                    </a>
                    <span v-else-if="column.type === 'link'">
                      <nuxt-link v-if="item.link" :to="item.link"
                        >{{ itemValue(item, column) }}
                      </nuxt-link>
                      <span v-else>{{ itemValue(item, column) }}</span>
                      <base-action-tooltip tooltip="Copied">
                        <base-button
                          title="Copy to clipboard"
                          class="table-info__actions__button button-icon"
                          @click.prevent="onActionClicked('copy-name', item)"
                        >
                          <svgicon name="copy" width="16" height="16" />
                        </base-button>
                      </base-action-tooltip>
                    </span>
                    <base-date
                      format="date-relative-now"
                      v-else-if="column.type === 'date'"
                      :date="itemValue(item, column)"
                    />
                    <span v-else-if="column.type === 'number'">
                      {{ itemValue(item, column) | formatNumber }}
                    </span>
                    <span
                      v-else-if="
                        !isNaN(itemValue(item, column)) &&
                        column.type === 'percentage'
                      "
                    >
                      {{ itemValue(item, column) | percent }}
                    </span>
                    <span v-else-if="column.type === 'array'">
                      <p
                        v-for="(arrayItem, index) in itemValue(item, column)"
                        :key="index"
                      >
                        {{ arrayItem
                        }}{{
                          index + 1 === itemValue(item, column).length
                            ? ""
                            : ","
                        }}
                      </p>
                    </span>
                    <span v-else-if="column.type === 'object'">
                      <p
                        v-for="key in Object.keys(itemValue(item, column))"
                        :key="key"
                      >
                        <strong>{{ key }}:</strong>
                        {{ itemValue(item, column)[key] }}
                      </p>
                    </span>
                    <!-- TODO: remove references to task -->
                    <span v-else-if="column.type === 'task'">
                      {{ itemValue(item, column) }}
                    </span>
                    <span v-else>{{ itemValue(item, column) }}</span>
                  </span>
                </span>
                <div v-if="visibleActions" class="table-info__actions">
                  <base-action-tooltip
                    v-for="action in filterActions"
                    :key="action.index"
                    :tooltip="action.tooltip"
                  >
                    <base-button
                      :title="action.title"
                      class="table-info__actions__button button-icon"
                      @click="onActionClicked(action.name, item)"
                    >
                      <svgicon
                        v-if="action.icon !== undefined"
                        :name="action.icon"
                        width="16"
                        height="16"
                      />
                    </base-button>
                  </base-action-tooltip>
                </div>
                <base-modal
                  v-if="!!deleteModalContent"
                  :modal-custom="true"
                  :prevent-body-scroll="true"
                  modal-class="modal-secondary"
                  :modal-title="deleteModalContent.title"
                  :modal-visible="visibleModalId === item.id"
                  @close-modal="$emit('close-modal')"
                >
                  <div>
                    <p v-html="deleteModalContent.text"></p>
                    <div class="modal-buttons">
                      <base-button
                        class="primary outline"
                        @click="$emit('close-modal')"
                      >
                        Cancel
                      </base-button>
                      <base-button
                        class="primary"
                        @click="onActionClicked('confirm-delete', item)"
                      >
                        Yes, delete
                      </base-button>
                    </div>
                  </div>
                </base-modal>
              </div>
            </li>
          </ul>
        </div>
      </template>
    </div>
  </transition>
</template>

<script>
import "assets/icons/trash-empty";
import "assets/icons/refresh";
import "assets/icons/copy";
import "assets/icons/link";
import "assets/icons/sort";
export default {
  props: {
    data: {
      type: Array,
      default: () => [],
    },
    actions: {
      type: Array,
      default: () => [],
    },
    columns: Array,
    searchOn: {
      type: String,
      default: undefined,
    },
    sortedOrder: {
      type: String,
      default: "desc",
    },
    sortedByField: {
      type: String,
      default: undefined,
    },
    noDataInfo: {
      title: undefined,
      message: undefined,
      icon: undefined,
    },
    emptySearchInfo: {
      title: undefined,
    },
    hideButton: {
      type: Boolean,
      default: false,
    },
    visibleModalId: {
      type: String | Array,
      default: undefined,
    },
    deleteModalContent: {
      type: Object,
    },
    globalActions: {
      type: Boolean,
      default: true,
    },
    querySearch: {
      type: String,
      default: undefined,
    },
    activeFilters: {
      type: Array,
      default: () => {
        return [];
      },
    },
  },
  data() {
    return {
      sortOrder: this.sortedOrder,
      visibleActions: true,
      sortedBy: this.sortedByField,
      allRecordsSelected: false,
      selectedItems: [],
      filters: {},
    };
  },
  computed: {
    tableIsEmpty() {
      return this.filteredResults && this.filteredResults.length === 0;
    },
    filterActions() {
      return this.actions.filter((a) => a.hide !== this.hideButton);
    },
    filteredResults() {
      const matchSearch = (item) => {
        if (this.querySearch === undefined) {
          return true;
        }
        const querySearch = this.querySearch.toLowerCase();
        if (this.searchOn) {
          return item[this.searchOn]
            .toString()
            .toLowerCase()
            .includes(querySearch);
        }
        return false;
      };
      const matchFilters = (item) => {
        if (this.filters) {
          return Object.keys(this.filters).every((key) => {
            if (this.isObject(item[key])) {
              return this.filters[key].find(
                (filter) => filter.value === item[key][filter.key]
              );
            } else {
              return this.filters[key].toString().includes(item[key]);
            }
          });
        }
        return true;
      };
      const itemComparator = (a, b) => {
        let modifier = 1;
        if (this.sortedOrder === "desc") modifier = -1;
        if (a[this.sortedBy] < b[this.sortedBy]) return -1 * modifier;
        if (a[this.sortedBy] > b[this.sortedBy]) return 1 * modifier;
        return 0;
      };
      const results = this.data.filter(matchSearch).filter(matchFilters);
      return results.sort(itemComparator);
    },
  },
  beforeMount() {
    this.sortedBy = this.sortedByField;
    const appliedFilters = this.activeFilters.filter(
      (filter) => filter.values.length
    );
    (appliedFilters || []).forEach(({ column, values }) => {
      this.$set(this.filters, column, values);
    });
  },
  methods: {
    isObject(obj) {
      return Object.prototype.toString.call(obj) === "[object Object]";
    },
    itemValue(item, column) {
      if (column.subfield) {
        return item[column.field][column.subfield];
      }
      return item[column.field];
    },
    onActionClicked(action, item) {
      this.$emit("onActionClicked", action, item);
    },
    sort(column) {
      this.$emit("sort-column", column.field, this.sortOrder);
      this.sortedBy = column.field;
      if (column.field === this.sortedBy) {
        this.sortOrder = this.sortOrder === "asc" ? "desc" : "asc";
      }
    },
    onApplyFilters(column, selectedOptions) {
      if (selectedOptions.length) {
        this.$set(this.filters, column.field, selectedOptions);
      } else {
        this.$delete(this.filters, column.field);
      }
      this.$emit("filter-applied", {
        column: column.field,
        values: selectedOptions,
      });
    },
    selectAll(value) {
      this.onAllCheckboxChanged(value);
    },
    onAllCheckboxChanged(value) {
      this.filteredResults.forEach((r) => {
        const rec = r;
        rec.selectedRecord = value;
      });
      this.selectedItems = this.filteredResults.filter(
        (f) => f.selectedRecord === true
      );
    },
    onCheckboxChanged(value, id) {
      this.filteredResults.forEach((r) => {
        if (r.id === id) {
          const rec = r;
          rec.selectedRecord = value;
        }
      });
      if (
        this.filteredResults.some(
          (f) => f.selectedRecord === false || undefined
        )
      ) {
        this.allRecordsSelected = false;
      } else {
        this.allRecordsSelected = true;
      }
      this.selectedItems = this.filteredResults.filter(
        (f) => f.selectedRecord === true
      );
    },
  },
};
</script>

<style lang="scss" scoped>
.table-info {
  $this: &;
  padding: 0;
  list-style: none;
  margin-bottom: 5em;
  color: $black-54;
  ul {
    list-style: none;
    padding: 0;
    margin: 0;
  }
  &__header {
    background: palette(white);
    min-height: 50px;
    position: relative;
    margin-top: $base-space * 2;
    &__checkbox {
      margin: 0 !important;
    }
    &__button {
      position: absolute !important;
      top: 2em;
      right: 2em;
      margin-bottom: 0 !important;
    }
    #{$this}__item {
      min-height: 50px;
      background: none;
      padding-top: 0.2em;
      padding-bottom: 0.2em;
      display: flex;
      align-items: center;
    }
    button:not(.button) {
      cursor: pointer;
      border: 0;
      outline: none;
      background: transparent;
      padding-left: 0;
      padding-right: 0;
      color: $black-87;
      @include font-size(14px);
      text-align: left;
      display: flex;
      align-items: center;
      @include media("<=desktop") {
        display: block;
        .svg-icon {
          display: block;
        }
      }
      span {
        white-space: nowrap;
      }
    }
  }
  &__body {
    overflow: auto;
    height: calc(100vh - 203px);
    padding-bottom: 0.5em;
    @extend %hide-scrollbar;
    #{$this}__item {
      margin-bottom: -1px;
      &:hover,
      &:focus {
        background: #fcfcfc;
      }
    }
  }
  &__item {
    background: palette(white);
    position: relative;
    list-style: none;
    padding: 2em 5em 2em 2em;
    display: flex;
    width: 100%;
    align-items: flex-start;
    text-decoration: none;
    outline: none;
    border: 1px solid palette(grey, 700);
    &__col {
      text-align: left;
      margin-right: 1em;
      flex: 1 1 0px;
      &:nth-last-of-type(-n + 1) {
        max-width: 120px;
      }
      &:first-child {
        width: auto;
        min-width: auto;
        flex-grow: 1.5;
      }
    }
    .svg-icon {
      margin-right: $base-space;
      fill: $black-54;
      &:hover {
        fill: $black-87;
      }
    }
  }
  &__tag {
    background: palette(grey, 100);
    display: inline-block;
    border-radius: $border-radius-s;
    color: palette(white);
    @include font-size(12px);
    box-shadow: 0 1px 4px 1px rgba(222, 222, 222, 0.5);
    padding: 0.1em 0.5em;
    margin-left: 1em;
  }
  .empty {
    margin-top: 5em;
    height: auto;
  }
  &__actions {
    position: absolute;
    right: 2em;
    &__button {
      position: relative;
      margin-left: 2em;
      padding: 0 !important;
      display: inline-block;
      .svg-icon {
        margin: 0;
      }
      & + #{$this} {
        margin-left: auto;
      }
    }
  }
  &__title {
    display: block;
    hyphens: auto;
    word-break: break-word;
    .button-icon {
      padding: $base-space;
      margin-left: $base-space;
      display: inline-block;
      .svg-icon {
        margin: 0;
      }
    }
    a {
      text-decoration: none;
      &:hover {
        color: palette(black);
      }
    }
  }
  .text {
    color: $black-54;
    p {
      display: inline-block;
      background: palette(grey, 800);
      padding: 0.5em;
      border-radius: 10px;
      margin-right: 0.5em;
      margin-top: 0;
      hyphens: auto;
      word-break: break-word;
      &:last-child {
        margin-bottom: 0;
      }
    }
  }
  .array {
    p {
      margin-top: 0;
      margin-bottom: 0;
      display: inline;
      hyphens: auto;
      word-break: break-word;
    }
  }

  // .metrics {
  //   display: block;
  //   span {
  //     display: inline-block;
  //     padding: 0.2em 0.5em;
  //     margin-top: 1em;
  //     background: lighten(palette(blue, 300), 44%);
  //     margin-right: 3px;
  //     border-radius: 2px;
  //   }
  // }
}
button[data-title] {
  position: relative;
  @extend %has-tooltip--top;
  @extend %tooltip-large-text;
}
</style>
