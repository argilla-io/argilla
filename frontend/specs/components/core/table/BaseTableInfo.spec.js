import { mount } from "@vue/test-utils";
import BaseTableInfo from "@/components/base/table/BaseTableInfo";

const mountBaseTableInfo = () =>
  mount(BaseTableInfo, {
    stubs: [
      "lazy-table-filtrable-column",
      "base-modal",
      "base-button",
      "base-modal",
    ],
    propsData: {
      actions: [],
      columns: [
        {
          idx: 1,
          key: "column1",
          class: "text",
          field: "workspace",
          filtrable: "true",
          name: "Workspace",
          type: "text",
        },
      ],
      data: [
        {
          id: "data1",
          name: "dataset_1",
          workspace: "recognai",
          task: "TokenClassification",
        },
        {
          id: "data2",
          name: "dataset_2",
          workspace: "recognai",
          task: "TokenClassification",
        },
      ],
      deleteModalContent: {
        text: "You are about to delete: <strong>undefined</strong>. This action cannot be undone",
        title: "Delete confirmation",
      },
      emptySearchInfo: {
        title: "0 datasets found",
      },
      globalActions: false,
      groupBy: undefined,
      hideButton: false,
      noDataInfo: undefined,
      querySearch: undefined,
      filterFromRoute: "workspace",
      searchOn: "name",
      sortedByField: "last_updated",
      sortedOrder: "desc",
      visibleModalId: undefined,
    },
    mocks: {
      $route: {
        query: {},
      },
    },
  });

describe("BaseTableInfo", () => {
  const spy = jest.spyOn(console, "error");
  afterEach(() => spy.mockReset());

  test("renders properly", () => {
    const wrapper = mountBaseTableInfo();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
