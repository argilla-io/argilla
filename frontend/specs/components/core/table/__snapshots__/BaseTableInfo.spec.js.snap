// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseTableInfo renders properly 1`] = `
<transition-stub name="fade" appear="true" filterfromroute="workspace">
  <div class="table-info">
    <div class="table-info__header">
      <div class="table-info__item">
        <div class="table-info__item__col">
          <lazy-table-filtrable-column-stub column="[object Object]" data="[object Object],[object Object]" filters="[object Object]"></lazy-table-filtrable-column-stub>
        </div>
      </div>
    </div>
    <div class="table-info__body">
      <ul>
        <li>
          <div class="table-info__item">
            <!----> <span class="table-info__item__col"><span class="text"><span>recognai</span></span></span>
            <div class="table-info__actions"></div>
            <base-modal-stub modal-custom="true" prevent-body-scroll="true" modal-class="modal-secondary" modal-title="Delete confirmation">
              <div>
                <p>You are about to delete: <strong>undefined</strong>. This action cannot be undone</p>
                <div class="modal-buttons">
                  <base-button-stub class="primary outline">
                    Cancel
                  </base-button-stub>
                  <base-button-stub class="primary">
                    Yes, delete
                  </base-button-stub>
                </div>
              </div>
            </base-modal-stub>
          </div>
        </li>
        <li>
          <div class="table-info__item">
            <!----> <span class="table-info__item__col"><span class="text"><span>recognai</span></span></span>
            <div class="table-info__actions"></div>
            <base-modal-stub modal-custom="true" prevent-body-scroll="true" modal-class="modal-secondary" modal-title="Delete confirmation">
              <div>
                <p>You are about to delete: <strong>undefined</strong>. This action cannot be undone</p>
                <div class="modal-buttons">
                  <base-button-stub class="primary outline">
                    Cancel
                  </base-button-stub>
                  <base-button-stub class="primary">
                    Yes, delete
                  </base-button-stub>
                </div>
              </div>
            </base-modal-stub>
          </div>
        </li>
      </ul>
    </div>
  </div>
</transition-stub>
`;
